From 5f5efc8f98d29b1bbc01afe3b649532e932d1cd3 Mon Sep 17 00:00:00 2001
From: Ryan Volz <ryan.volz@gmail.com>
Date: Sun, 7 Mar 2021 14:15:02 -0500
Subject: [PATCH 4/4] cmake: Format numpy include path in CMake style for
 consistency.

GNU Radio builds done with conda do path subsitution to replace the
build prefix with the installed prefix, but it struggles when multiple
path separator styles are used in the same file. This makes it so that
all of the paths that would be subsitituted use the same style (forward
slash separators) so that build prefix replacement happens correctly.
---
 CMakeLists.txt                                      | 3 +++
 gr-utils/modtool/templates/gr-newmod/CMakeLists.txt | 3 +++
 2 files changed, 6 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 83f15f06c..d0e166ebb 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -352,6 +352,9 @@ execute_process(
     COMMAND "${PYTHON_EXECUTABLE}" -c
     "try:\n import numpy\n import os\n inc_path = numpy.get_include()\n if os.path.exists(os.path.join(inc_path, 'numpy', 'arrayobject.h')):\n  print(inc_path, end='')\nexcept:\n pass"
     OUTPUT_VARIABLE PYTHON_NUMPY_INCLUDE_DIR)
+# format path in CMake-style for consistency with other path variables
+# (a consistent style helps conda builds by using the same path separators)
+file(TO_CMAKE_PATH "${PYTHON_NUMPY_INCLUDE_DIR}" PYTHON_NUMPY_INCLUDE_DIR)
 
 include(GrComponent)
 GR_REGISTER_COMPONENT("python-support" ENABLE_PYTHON
diff --git a/gr-utils/modtool/templates/gr-newmod/CMakeLists.txt b/gr-utils/modtool/templates/gr-newmod/CMakeLists.txt
index 2e1daf173..843eea36f 100644
--- a/gr-utils/modtool/templates/gr-newmod/CMakeLists.txt
+++ b/gr-utils/modtool/templates/gr-newmod/CMakeLists.txt
@@ -124,6 +124,9 @@ execute_process(
     COMMAND "${PYTHON_EXECUTABLE}" -c
     "try:\n import numpy\n import os\n inc_path = numpy.get_include()\n if os.path.exists(os.path.join(inc_path, 'numpy', 'arrayobject.h')):\n  print(inc_path, end='')\nexcept:\n pass"
     OUTPUT_VARIABLE PYTHON_NUMPY_INCLUDE_DIR)
+# format path in CMake-style for consistency with other path variables
+# (a consistent style helps conda builds by using the same path separators)
+file(TO_CMAKE_PATH "${PYTHON_NUMPY_INCLUDE_DIR}" PYTHON_NUMPY_INCLUDE_DIR)
 
 ########################################################################
 # Setup doxygen option
-- 
2.25.1

